src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -399,13 +765,97 @@ static ObjectInputFilter createFilter2(String pattern) {
             return Global.createFilter(pattern, false);
         }
 
+        /**
+         * Returns a filter that returns {@code Status.ALLOWED} if the predicate on the class is {@code true},
+         * otherwise the {@code otherStatus}.
```

Comment:
I originally overlooked the fact that UNDECIDED can be returned by these filters. Would it be clearer to drop "otherwise the otherStatus" ?? I also wonder if otherStatus carries its own weight? How useful is it to return an otherStatus that is not UNDECIDED?

Reply:
A filter author may prefer to return REJECTED if the predicate fails.
It gives the developer control on when UNDECIDED values can be part of a filter expression.